test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -461,93 +495,398 @@ TEST(TypeObjectUtilsTests, build_plain_map_s_type_defn_inconsistencies)
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_typeid, flags, complete_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_typeid, flags, complete_indirect_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_typeid, flags, complete_alias_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_typeid, flags, minimal_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_typeid, flags, minimal_indirect_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_typeid, flags, minimal_alias_id), InvalidArgumentError);
+
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, key_identifier), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, complete_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, complete_indirect_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, complete_alias_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, minimal_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, minimal_indirect_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_indirect_id, flags, minimal_alias_id), InvalidArgumentError);
+
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, complete_alias_id, flags, key_identifier));
```

Comment:
Again, a fully descriptive header cannot have an alias. Consider this criteria for the rest of the tests, please.

Reply:
Changed aliases behabiour in [d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)